### PR TITLE
kvm: replace systemd-nspwan with lkvm

### PR DESCRIPTION
**Pull request by ppalucki**
WIP: do not merge, RFC

creates additional flavor with kernel/lkvm (kvmtool) included and
calls lkvm instead a systemd-nspwan

includes:
- kernel config and optimization patches, Makefile
- lkvm and Makefile
- slightly modified init to run lkvm
- docs

https://github.com/coreos/rkt/pull/1115

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -86,6 +89,7 @@ if [[ "$(uname)" == "Linux" ]]; then
 	if [ "$RKT_STAGE1_USR_FROM" != "none" ] ; then
 		JOBS=1
 		which lscpu >&/dev/null && JOBS=$(lscpu --parse=CPU | grep -c ^[^#])
+		JOBS=1
```
Comment (ppalucki) [inline build]: to remove

File: stage1/rootfs/aggregate/Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ S1=stage1
 S1ACI=../../../bin/$(S1).aci
 
 $(S1ACI): aggregate.sh Makefile scripts/* units/* install.d/* aci-manifest
-	@./aggregate.sh && cp aci-manifest $(S1)/manifest && ${ACTOOL} build --overwrite $(S1) $(S1ACI)
+	@./aggregate.sh && cp aci-manifest $(S1)/manifest && ${ACTOOL} --debug build --no-compression --overwrite $(S1) $(S1ACI)
```
Comment (ppalucki) [inline stage1/rootfs/aggregate/Makefile]: to remove

File: build [hunk 3]
```diff
@@ -30,6 +30,9 @@ case "${RKT_STAGE1_USR_FROM}" in
 		export RKT_STAGE1_SYSTEMD_VER="${RKT_STAGE1_SYSTEMD_VER:-v220}"
 		echo "stage1 will be built from sources: $RKT_STAGE1_SYSTEMD_SRC $RKT_STAGE1_SYSTEMD_VER"
 		;;
+    kvm)
```
Comment (jellonek) [inline build]: wrong indent level

File: build [hunk 4]
```diff
@@ -30,6 +30,9 @@ case "${RKT_STAGE1_USR_FROM}" in
...
+		echo "stage1 will reuse coreos image with additional hypervisor components"
+        ;;
```
Comment (jellonek) [inline build]: wrong indent level

File: stage1/rootfs/usr_from_kvm/kernel/Makefile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+KERNEL_PATH=https://www.kernel.org/pub/linux/kernel/v4.x/
+
+bzImage: $(KERNEL_DIR)/Makefile
+	./patches.sh $(KERNEL_DIR)
+	cp cutdown-config $(KERNEL_DIR)/.config
+	(cd $(KERNEL_DIR) ; make -j`nproc` bzImage)
+	cp $(KERNEL_DIR)/arch/x86/boot/bzImage .
+
+$(KERNEL_DIR)/Makefile: $(KERNEL_FILE)
+	tar xf $(KERNEL_FILE)
```
Comment (jellonek) [inline stage1/rootfs/usr_from_kvm/kernel/Makefile:14]: whitespace at end of line

**Comment by ppalucki** [comment]
```
export RKT_STAGE1_USR_FROM=kvm
./build 
sudo ./bin/rkt --insecure-skip-verify run --volume=database,kind=empty ../ace-validator-main.aci ../ace-validator-sidekick.aci
```

result

```
[    0.213983] ace-validator[75]: prestart OK
[    5.214098] ace-validator[74]: sidekick FAIL
[    5.214382] ace-validator[74]: ==> error touching file "/db/sidekick": open /db/sidekick: no such file or directory
[    5.214573] ace-validator[74]: ==> timed out waiting for /db/main
[    5.228179] ace-validator[80]: main FAIL
[    5.228362] ace-validator[80]: ==> error calling statfs on "/db": no such file or directory
[    5.228517] ace-validator[80]: ==> timed out waiting for /db/sidekick
[    5.228663] ace-validator[80]: ==> error touching file "/db/main": open /db/main: no such file or directory

```

**Pull request closed without merging**